notify/metricslog_test.go

```diff
@@ -223,6 +223,111 @@ func TestRestart(t *testing.T) {
 	checkIndex(t, checkpoint, 6)
 }
 
+func TestRemoveOldestSamples(t *testing.T) {
+	tests := []struct {
+		name           string
+		log            *MetricsLog
+		samplesToWrite []uint    // Samples to write log before removal
+		numSamples     int       // Number of oldest samples to remove
+		wantLogFile    []float64 // Expected remaining samples after removal
+		wantIndex      uint64    // Expected index after removal
+		wantErr        bool
+		wantErrMsg     string
+	}{
+		{
+			name:           "Remove 3 oldest samples",
+			log:            &MetricsLog{path: "/tmp/test_log"},
```

Comment:
As all of these tests are executed in sequence this will work just fine.
If you want to ensure that the same tmp file isn't used in other tests that might lead to flanky test results you can use [ioutils.tempFile](https://pkg.go.dev/io/ioutil#example-TempFile)

Reply:
For this usecase it might be good to use `t.TempDir` https://pkg.go.dev/testing#B.TempDir as it is used e.g. in function `createMetricsPath` - or use this function.

Reply:
Thanks for the feedback. I think that's a good suggestion for this usecase tried updating the code, it's a little cleaner now too!